src/main.rs

```diff
@@ -124,7 +124,9 @@ fn main() -> Result<()> {
                 .context("SO_BINDTODEVICE failed")?;
             #[cfg(target_vendor = "apple")]
             socket
-                .bind_device_by_index(Some(libc::if_nametoindex(int.name.as_bytes())))
+                .bind_device_by_index(std::num::NonZeroU32::new(unsafe {
```

Comment:
Pardon my ignorance, but curious if simply `.bind_device_by_index(NonZeroU32::new(int.index))` will work here instead?

Reply:
Yup, missed the index field, thanks for taking a look!